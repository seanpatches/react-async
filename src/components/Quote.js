import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote }){
  return (
    <>
    <p>{quote.character}</p> 
    <img src={Quote.image}/>
    <p>{quote.quote}</p>
    </>
  ); 
}

Quote.propTypes = {
  quote: PropTypes.shape({
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    image: PropTypes.string
  })
};

export default Quote;
