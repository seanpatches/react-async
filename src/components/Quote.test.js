import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders a Quote', () => {
    const quote = {
      character: 'Fry',
      image: '/image',
      quote: 'What??'
    };
    const wrapper = shallow(<Quote quote={quote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
