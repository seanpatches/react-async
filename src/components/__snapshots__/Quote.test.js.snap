// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders a Quote 1`] = `
<Fragment>
  <p>
    Fry
  </p>
  <img />
  <p>
    What??
  </p>
</Fragment>
`;
